.github/searchResults.js

```diff
@@ -0,0 +1,114 @@
+class SearchResults {
+  constructor(container) {
+    this.container = container;
+    this.resultsList = this.container.querySelector("#searchResults");
+    this.theLoader = document.getElementById("loader");
+    this.baseUrl =
+      "https://stock-exchange-dot-full-stack-course-services.ew.r.appspot.com/api/v3/";
+    this.timeout = null;
+    this.currentSearchTerm = "";
+  }
+  async search(searchTerm) {
+    this.currentSearchTerm = searchTerm;
+    this.enableSpinner();
+    const endpoint = `${this.baseUrl}search?query=${searchTerm}&limit=10&exchange=NASDAQ`;
+
+    // Save search term in query string params
+    const params = new URLSearchParams(window.location.search);
+    params.set("query", searchTerm);
+    const newUrl = window.location.pathname + "?" + params.toString();
+    window.history.pushState({}, "", newUrl);
+
+    try {
+      const response = await fetch(endpoint);
+      const data = await response.json();
+      this.renderResults(data);
+      return data;
+    } catch (error) {
+      console.log("Error from server");
+    }
+  }
+  async renderResult(result) {
+    const listItem = document.createElement("li");
+    const symbol = result.symbol;
+    const name = result.name;
+    const logoUrl = await this.getLogoUrl(symbol);
+    const stockChange = await this.getStockChange(symbol);
+    const stockChangeColor = this.getStockChangeColor(stockChange);
+    const searchTerm = this.currentSearchTerm.toLowerCase();
+
+    const symbolIndex = symbol.toLowerCase().indexOf(searchTerm);
+    const nameIndex = name.toLowerCase().indexOf(searchTerm);
+
+    const symbolStart = symbol.substring(0, symbolIndex);
+    const symbolMatch = symbol.substring(
+      symbolIndex,
+      symbolIndex + searchTerm.length
+    );
+    const symbolEnd = symbol.substring(symbolIndex + searchTerm.length);
+
+    const nameStart = name.substring(0, nameIndex);
```

Comment:
nice. a better way to do milestone 7 would be to create a function that highlights the relevant text, and use it wherever you need. DRY